Caused by: java.lang.NoClassDefFoundError: org/grails/web/sitemesh/GroovyPageLayoutFinder

### Expected Behavior

The command `./gradlew compileGroovy` should execute successfully without any errors. 

### Actual Behaviour

The task `compileGroovy` fail with the following exception:


```
Caused by: java.lang.NoClassDefFoundError: org/grails/web/sitemesh/GroovyPageLayoutFinder
        at java.desktop/com.sun.beans.introspect.MethodInfo.get(MethodInfo.java:70)
        at java.desktop/com.sun.beans.introspect.ClassInfo.getMethods(ClassInfo.java:80)
        at org.grails.plugins.web.interceptors.InterceptorArtefactHandler.<clinit>(InterceptorArtefactHandler.groovy:33)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at org.grails.core.io.support.GrailsFactoriesLoader.loadFactoriesWithArguments(GrailsFactoriesLoader.groovy:65)
        at org.grails.core.io.support.GrailsFactoriesLoader.loadFactories(GrailsFactoriesLoader.groovy:47)
        at org.grails.compiler.injection.GlobalGrailsClassInjectorTransformation.visit(GlobalGrailsClassInjectorTransformation.groovy:54)
        at org.gradle.api.internal.tasks.compile.ApiGroovyCompiler.execute(ApiGroovyCompiler.java:285)
        at org.gradle.api.internal.tasks.compile.ApiGroovyCompiler.execute(ApiGroovyCompiler.java:67)
        at org.gradle.api.internal.tasks.compile.GroovyCompilerFactory$DaemonSideCompiler.execute(GroovyCompilerFactory.java:97)
        at org.gradle.api.internal.tasks.compile.GroovyCompilerFactory$DaemonSideCompiler.execute(GroovyCompilerFactory.java:76)
        at org.gradle.api.internal.tasks.compile.daemon.AbstractDaemonCompiler$CompilerWorkAction.execute(AbstractDaemonCompiler.java:135)
        at org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:63)
        at org.gradle.workers.internal.AbstractClassLoaderWorker$1.create(AbstractClassLoaderWorker.java:49)
        at org.gradle.workers.internal.AbstractClassLoaderWorker$1.create(AbstractClassLoaderWorker.java:43)
        at org.gradle.internal.classloader.ClassLoaderUtils.executeInClassloader(ClassLoaderUtils.java:100)
        at org.gradle.workers.internal.AbstractClassLoaderWorker.executeInClassLoader(AbstractClassLoaderWorker.java:43)
        at org.gradle.workers.internal.IsolatedClassloaderWorker.run(IsolatedClassloaderWorker.java:49)
        at org.gradle.workers.internal.IsolatedClassloaderWorker.run(IsolatedClassloaderWorker.java:30)
        at org.gradle.workers.internal.WorkerDaemonServer.run(WorkerDaemonServer.java:87)
        at org.gradle.workers.internal.WorkerDaemonServer.run(WorkerDaemonServer.java:56)
        at org.gradle.process.internal.worker.request.WorkerAction$1.call(WorkerAction.java:138)
        at org.gradle.process.internal.worker.child.WorkerLogEventListener.withWorkerLoggingProtocol(WorkerLogEventListener.java:41)
        at org.gradle.process.internal.worker.request.WorkerAction.run(WorkerAction.java:135)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
        at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
        at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
        at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:49)

```

### Steps To Reproduce

1. Create a New Grails Web Application
2. Execute `./gradlew compileGroovy` 

### Environment Information

- macOS
- Java 11 

### Example Application

_No response_

### Version

6.0.0-SNAPSHOT

Please check the build log of [Grails Functional Tests](https://github.com/grails/grails-functional-tests) at https://github.com/grails/grails-functional-tests/actions/runs/6528756303/job/17725395107

--- a/grails-plugin-controllers/build.gradle
+++ b/grails-plugin-controllers/build.gradle
@@ -6,11 +6,9 @@ dependencies {
             project(':grails-plugin-domain-class')
 
     api("org.springframework.boot:spring-boot-autoconfigure:$springBootVersion")
-    compileOnly "org.grails:grails-web-sitemesh:$gspVersion"
-
+    api "org.grails:grails-web-sitemesh:$gspVersion"
     runtimeOnly project(':grails-plugin-i18n')
 
     testRuntimeOnly "jline:jline:$jlineVersion"
     testRuntimeOnly "org.fusesource.jansi:jansi:$jansiVersion"
-    testImplementation "org.grails:grails-web-sitemesh:$gspVersion"
 }

--- a/grails-plugin-interceptors/build.gradle
+++ b/grails-plugin-interceptors/build.gradle
@@ -1,6 +1,4 @@
 dependencies {
     api project(":grails-plugin-controllers")
     api project(":grails-plugin-url-mappings")
-    compileOnly "org.grails:grails-web-sitemesh:$gspVersion"
-    testImplementation "org.grails:grails-web-sitemesh:$gspVersion"
 }

--- a/grails-plugin-rest/build.gradle
+++ b/grails-plugin-rest/build.gradle
@@ -7,10 +7,8 @@ dependencies {
             project(":grails-plugin-datasource")
 
     api "org.grails.plugins:converters:$legacyConvertersVersion"
-    compileOnly "org.grails:grails-web-sitemesh:$gspVersion"
 
     implementation "com.github.ben-manes.caffeine:caffeine:$caffeineVersion"
     testImplementation project(":grails-plugin-url-mappings"),
                 project(":grails-test-suite-base")
-    testImplementation "org.grails:grails-web-sitemesh:$gspVersion"
 }

--- a/grails-web-boot/build.gradle
+++ b/grails-web-boot/build.gradle
@@ -3,7 +3,6 @@ dependencies {
 
     testImplementation project(":grails-plugin-controllers")
 
-    testImplementation "org.grails:grails-web-sitemesh:$gspVersion"
     testImplementation "org.apache.tomcat.embed:tomcat-embed-core:$tomcatVersion"
     testRuntimeOnly "org.apache.tomcat.embed:tomcat-embed-logging-juli:$tomcatLog4jVersion"
     testRuntimeOnly project(":grails-plugin-i18n")

